src/Type/Integer.php

```diff
@@ -0,0 +1,56 @@
+<?php
+/**
+ * Integer value object class.
+ *
+ * Implemented as an immutable object with a pair of named constructors.
+ */
+
+namespace Joomla\Webservices\Type;
+
+final class Integer extends Type
```

Comment:
I would go for Integer since Int is reserved class name in PHP7. ref: http://php.net/manual/en/reserved.other-reserved-words.php